src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -31,30 +37,45 @@
     private BodyExecution body;
     private transient ScheduledFuture<?> killer;
 
+    private long timeout = 0;
     private long end = 0;
+
+    /** Used to track whether this is timing out on inactivity without needing to reference {@link #step}. */
+    private boolean activity = false;
+
     /** whether we are forcing the body to end */
     private boolean forcible;
 
     TimeoutStepExecution(TimeoutStep step, StepContext context) {
         super(context);
         this.step = step;
+        this.activity = step.isActivity();
     }
 
     @Override
     public boolean start() throws Exception {
         StepContext context = getContext();
-        body = context.newBodyInvoker()
-                .withCallback(new Callback())
-                .start();
-        long now = System.currentTimeMillis();
-        end = now + step.getUnit().toMillis(step.getTime());
-        setupTimer(now);
+        BodyInvoker bodyInvoker = context.newBodyInvoker()
+                .withCallback(new Callback());
+
+        if (step.isActivity()) {
+            bodyInvoker = bodyInvoker.withContext(
+                    BodyInvoker.mergeConsoleLogFilters(
+                            context.get(ConsoleLogFilter.class),
+                            new ConsoleLogFilterImpl(new ResetCallbackImpl())
+                    )
+            );
+        }
+
+        body = bodyInvoker.start();
+        timeout = step.getUnit().toMillis(step.getTime());
+        resetTimer();
         return false;   // execution is asynchronous
     }
 
     @Override
     public void onResume() {
-        setupTimer(System.currentTimeMillis());
+        setupTimer(System.currentTimeMillis(), false);
```

Comment:
@abayer I wonder a bit about what the implications are for not resetting activity-based timers at resume... if we're looking at logs, won't the timer trip if Jenkins has been down for a while, and it takes a little bit to get the new log data?

Then again we also trip normal timers at restart too I guess.

Reply:
Yeah, I'm not sure what the ideal approach here would be, so I'd say let's go with this to start and change it in the future if needed. =)

Reply:
I have found issue with timeout activity, described in:
https://issues.jenkins.io/browse/JENKINS-58752

How additional logs could be taken from Jenkins, to investigate that issue?